netlink_darwin.go

```diff
@@ -0,0 +1,208 @@
+// +build darwin
+
+package main
+
+import (
+	"errors"
+	"fmt"
+	"log"
+	"net"
+	"syscall"
+	"unsafe"
+
+	"golang.org/x/net/route"
+	"golang.org/x/sys/unix"
+)
+
+type netlinkHandle struct {
+	localAddr net.IP
+}
+
+// NewNetLinkHandle will create a new NetLinkHandle
+func NewNetLinkHandle() *netlinkHandle {
+	return &netlinkHandle{}
+}
+
+// AddrReplace: will replace (or, if not present, add) an IP address on a link
+// device.
+func (h *netlinkHandle) UpdateIP(devName string, ipnet *net.IPNet) error {
+	fd, err := unix.Socket(unix.AF_INET, unix.SOCK_DGRAM, unix.AF_UNSPEC)
+	if err != nil {
+		return err
+	}
+	defer func() {
+		if err := unix.Close(fd); err != nil {
+			log.Printf("Could not close AF_INET socket: %v", err)
+		}
+	}()
+	if err := flushAddresses(fd, devName); err != nil {
```

Comment:
I am not sure you necessarily need to flush all addresses here. I've deleted that part from the linux file as I wanted to try multiple peers on the same interface. Assuming that the interface is created via the agent and dies with it (again as linux behaviour) this won't be needed for cleanup purposes(?)

Reply:
Yes, I wasn't sure how we want to handle this and whether we need Replace, Update or Add or potentially a combination of these and we'll need to refine this bit here.

For the initial setup, when the agent starts, we don't need to flush, it's already empty. If the agent is long running and we get a different allocation, we probably want to: add new IP, setup new routes, remove old IP (just the one) which I _think_ will also remove the old routes. This will likely end up in a smooth transition (more testing needed).

I haven't tested multiple peers on the same interface at all, but if it works as expected, we could just remove `h.localAddr` which is going to be the old address for a peer (assuming netlinkHandle and peers have a 1-to-1 relation).